File: services/clinical-service/src/domain/events/PatientAdmittedEvent.ts
import { DomainEvent } from './EncounterCreatedEvent';

export class PatientAdmittedEvent implements DomainEvent {
    readonly eventType = 'patient.admitted';
    readonly occurredAt: Date;
    readonly payload: Record<string, unknown>;

    constructor(encounterId: string, patientId: string) {
        this.occurredAt = new Date();
        this.payload = { encounterId, patientId };
    }
}

File: services/clinical-service/src/main/java/com/medtrust/clinical/domain/event/PatientAdmittedEvent.java
package com.medtrust.clinical.domain.event;

import java.time.Instant;
import java.util.Map;

public record PatientAdmittedEvent(
        String encounterId,
        String patientId) implements DomainEvent {

    @Override
    public String eventType() {
        return "patient.admitted";
    }

    @Override
    public Instant occurredAt() {
        return Instant.now();
    }

    @Override
    public Map<String, Object> payload() {
        return Map.of(
                "encounterId", encounterId,
                "patientId", patientId);
    }
}
